DemoApp-Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/HomeActivity.kt

```diff
@@ -1,17 +1,68 @@
 package com.applovin.enterprise.apps.demoapp
 
+import android.content.Intent
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.DefaultItemAnimator
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.applovin.enterprise.apps.demoapp.ads.ProgrammaticBannerAdActivity
+import com.applovin.enterprise.apps.demoapp.ads.InterstitialAdActivity
+import com.applovin.enterprise.apps.demoapp.ads.LayoutEditorBannerAdActivity
+import com.applovin.enterprise.apps.demoapp.ads.RewardedAdActivity
+import com.applovin.enterprise.apps.demoapp.data.home.AdType
+import com.applovin.enterprise.apps.demoapp.data.home.ListItem
+import com.applovin.enterprise.apps.demoapp.data.home.SectionHeader
 import com.applovin.enterprise.apps.demoapp.kotlin.R
-
+import com.applovin.enterprise.apps.demoapp.ui.HomeRecyclerViewAdapter
+import com.applovin.sdk.AppLovinMediationProvider
+import com.applovin.sdk.AppLovinSdk
 import kotlinx.android.synthetic.main.activity_home.*
+import kotlinx.android.synthetic.main.content_home.*
 
-class HomeActivity : AppCompatActivity() {
+class HomeActivity : AppCompatActivity(),
+    HomeRecyclerViewAdapter.OnHomeListItemClickListener
+{
+    private val homeListItems: List<ListItem> by lazy {
+        listOf(
+            SectionHeader("MAX Ads"),
+            AdType("Interstitial", Intent(this, InterstitialAdActivity::class.java)),
+            AdType("Rewarded", Intent(this, RewardedAdActivity::class.java)),
+            AdType("Programmatic Banners / Leaders", Intent(this, ProgrammaticBannerAdActivity::class.java)),
+            AdType("Layout Editor Banners / Leaders", Intent(this, LayoutEditorBannerAdActivity::class.java))
+        )
+    }
 
-    override fun onCreate(savedInstanceState: Bundle?) {
+    override fun onCreate(savedInstanceState: Bundle?)
+    {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_home)
         setSupportActionBar(toolbar)
+
+        val recyclerViewAdapter = HomeRecyclerViewAdapter(homeListItems, this, this)
+        val manager = LinearLayoutManager(this)
+        val decoration = DividerItemDecoration(this, manager.orientation)
+
+        homeRecyclerView.apply {
+            setHasFixedSize(true)
+            layoutManager = manager
+            addItemDecoration(decoration)
+            itemAnimator = DefaultItemAnimator()
+            adapter = recyclerViewAdapter
+        }
+
+        // Initialize the AppLovin SDK
+        AppLovinSdk.getInstance(this).mediationProvider = AppLovinMediationProvider.MAX
```

Comment:
Were we initializing this in a custom `Application` class in Java? It's a bit more explicit doing it like that, whereas here the SDK's initialization is a bit hidden.

Reply:
I did the same thing in Java, initializing the SDK in the `onCreate` for the `HomeActivity`.

Reply:
It might be easier for pubs to see it if you re-name it to `MainActivity` (sinec most docs ask to initialize in Applicaiton class or "your main activity")